climakitae/threshold_tools.py

```diff
@@ -0,0 +1,613 @@
+
+#####################################################################
+
+
+def get_ams(da, extremes_type="max"):
+
+    extremes_types = ["max"]
+    if extremes_type not in extremes_types:
+        raise ValueError(
+            "invalid extremes type. expected one of the following: %s" % extremes_types
+        )
+
+    if extremes_type == "max":
+        ams = da.resample(time="A").max(keep_attrs=True)
+        ams.attrs["extreme value extraction method"] = "block maxima"
+        ams.attrs["extremes type"] = "maxima"
+        ams.attrs["block size"] = "1 year"
+        ams.attrs["timeseries type"] = "annual max series"
+
+    # if extremes_type == 'min':
+    #     ams = da.resample(time='A').min(keep_attrs=True)
+    #     ams.attrs['extreme value extraction method'] = 'block maxima'
+    #     ams.attrs['extremes type'] = 'minima'
+    #     ams.attrs['block size'] = '1 year'
+    #     ams.attrs['timeseries type'] = 'annual min series'
+
+    return ams
+
+
+#####################################################################
+
+
+def get_lmoments(ams, distr="gev", multiple_points=True):
+
+    distrs = ["gev", "gumbel", "weibull", "pearson3", "genpareto"]
+    if distr not in distrs:
+        raise ValueError(
+            "invalid distr type. expected one of the following: %s" % distrs
+        )
+
+    ams_attributes = ams.attrs
+
+    if multiple_points == True:
+        ams = ams.stack(allpoints=["x", "y"]).squeeze().groupby("allpoints")
+
+    def apply_function(ams):
+        def function(ams):
+
+            if distr == "gev":
+                lmoments = ldistr.gev.lmom_fit(ams)
+
+            if distr == "gumbel":
+                lmoments = ldistr.gum.lmom_fit(ams)
+
+            if distr == "weibull":
+                lmoments = ldistr.wei.lmom_fit(ams)
+
+            if distr == "pearson3":
+                lmoments = ldistr.pe3.lmom_fit(ams)
+
+            if distr == "genpareto":
+                lmoments = ldistr.gpa.lmom_fit(ams)
+
+            return lmoments
+
+        return xr.apply_ufunc(
+            function,
+            ams,
+            input_core_dims=[["time"]],
+            exclude_dims=set(("time",)),
+            output_core_dims=[[]],
+        )
+
+    lmoments = apply_function(ams)
+
+    new_ds = lmoments.to_dataset()
+    new_ds = new_ds.rename({"T2": "lmoments"})
+
+    if multiple_points == True:
+        new_ds = new_ds.unstack("allpoints")
+
+    new_ds.attrs = ams_attributes
+    new_ds.attrs["distribution"] = "{}".format(str(distr))
+
+    return new_ds
+
+
+#####################################################################
+
+
+def get_aicc_stat(ams, multiple_points=True):
+
+    ams_attributes = ams.attrs
+
+    if multiple_points == True:
+        ams = ams.stack(allpoints=["x", "y"]).squeeze().groupby("allpoints")
+
+    def apply_function(ams):
+        def function(ams):
+
+            try:
+                lmoments_gev = ldistr.gev.lmom_fit(ams)
```

Comment:
It would help readability here by looping over the distribution names and calling `AICc` on each one instead of multiple lines with the same call.